src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/file.py

```diff
@@ -7,65 +7,57 @@
 from azure.common import AzureException, AzureHttpError
-from azure.cli.core.profiles import supported_api_version, ResourceType
+from azure.cli.core.profiles import supported_api_version, ResourceType, get_sdk
 from azure.cli.command_modules.storage.util import (filter_none, collect_blobs, collect_files,
                                                     create_blob_service_from_storage_client,
                                                     create_short_lived_container_sas,
-                                                    create_short_lived_share_sas)
+                                                    create_short_lived_share_sas, guess_content_type)
 
 
-def storage_file_upload_batch(client, destination, source, pattern=None, dryrun=False,
-                              validate_content=False, content_settings=None, max_connections=1,
-                              metadata=None):
-    """
-    Upload local files to Azure Storage File Share in batch
-    """
+def storage_file_upload_batch(client, destination, source, pattern=None, dryrun=False, validate_content=False,
+                              content_settings=None, max_connections=1, metadata=None):
+    """ Upload local files to Azure Storage File Share in batch """
 
     from .util import glob_files_locally
     source_files = [c for c in glob_files_locally(source, pattern)]
+    logger = get_az_logger(__name__)
+    settings_class = get_sdk(ResourceType.DATA_STORAGE, 'file.models#ContentSettings')
 
     if dryrun:
-        logger = get_az_logger(__name__)
-        logger.warning('upload files to file share')
-        logger.warning('    account %s', client.account_name)
-        logger.warning('      share %s', destination)
-        logger.warning('      total %d', len(source_files or []))
-        logger.warning(' operations')
-        for f in source_files or []:
-            logger.warning('  - %s => %s', *f)
-
-        return []
+        return [{'File': client.make_file_url(destination, os.path.dirname(src), os.path.basename(dst)),
+                 'Type': guess_content_type(src, content_settings, settings_class).content_type}
+                for src, dst in source_files]
 
     # TODO: Performance improvement
     # 1. Upload files in parallel
-    def _upload_action(source_pair):
-        dir_name = os.path.dirname(source_pair[1])
-        file_name = os.path.basename(source_pair[1])
+    def _upload_action(src, dst):
+        dir_name = os.path.dirname(dst)
+        file_name = os.path.basename(dst)
 
         _make_directory_in_files_share(client, destination, dir_name)
         create_file_args = {
             'share_name': destination,
             'directory_name': dir_name,
             'file_name': file_name,
-            'local_file_path': source_pair[0],
-            'content_settings': content_settings,
+            'local_file_path': src,
+            'content_settings': guess_content_type(src, content_settings, settings_class),
```

Comment:
It looks like this test is using the guess content type stuff, but I don't see where you verify that it guessed correctly.

Reply:
Per discussion, I understand that this can't be reliably done.